executor/executor.go

```diff
@@ -1931,3 +1931,49 @@ func (e *DummyScanExec) Close() error {
 func (e *DummyScanExec) Next() (*Row, error) {
 	return nil, nil
 }
+
+// TempStoreExec represents TempStore executor.
+// it stores the return values of the executor of its child node.
+type TempStoreExec struct {
+	schema     expression.Schema
+	Src        Executor
+	storedRows []*Row
+	cursor     int
+	closed     bool
+}
+
+// Schema implements the Executor Schema interface.
+func (e *TempStoreExec) Schema() expression.Schema {
+	return e.schema
+}
+
+// Close implements the Executor Close interface.
+func (e *TempStoreExec) Close() error {
+	e.cursor = 0
+	return nil
+}
+
+// Next implements the Executor Next interface.
+func (e *TempStoreExec) Next() (*Row, error) {
+	if e.closed {
+		e.cursor++
```

Comment:
In all other executors, we update cursor after getting a row, we better keep it consistent.